### ISSUE TITLE
Q: Why does it abort when encryption is enabled

### ISSUE THREAD

**Issue by Slevin90** [rullzer/previewgenerator#47]
Hi,

The idea of the app is very good.
why can't previewgenerator work if encryption is enabled?
To make Nextcloud secure, https is the good option...

Thanks for the answer

**Comment by Slevin90** [comment]
I'd like to add to this;
https://docs.nextcloud.com/server/11/admin_manual/configuration_files/encryption_configuration.html

In my particular case;  i only have file encryption enabled for external storage.
Home storage is not encrypted.

Maybe the check about encryption enabled could check for this specific setting in stead of overall encryption:enabled ?

**Comment by rullzer** [comment]
Because encryption is just to much stuff to check and handle right now. I might look into it in the future but right now it is safer to just not do it.

**Comment by joekerna** [comment]
I have disabled encryption but the app is still saying

> Encryption is enabled. Aborted.

**Comment by neckflo** [comment]
I have the same Problem. I have no shared Files( i have deleted all shares), Encryption is disabled. And i get this error.

[OC\Encryption\Exceptions\DecryptionFailedException]
  Can not decrypt this file, probably this is a shared file. Please ask the file owner to reshare the file with you.

**Comment by mmccarn** [comment]
With Nextcloud 12.0.3, it doesn't seem possible to completely disable server-side encryption from the GUI.

The 'Enable server-side encryption' checkbox in admin->encryption remains checked even if the encryption app is disabled -- and does not seem to be un-checkable (although I did not try unchecking it with the encryption app _enabled_).

To disable server-side encryption, use occ:
```
sudo -u www-data php /var/www/nextcloud/occ encryption:disable
```

**Comment by rullzer** [comment]
Long story short. I don't want to handle all the encryption use cases. So lets close this.

**Comment by bcutter** [comment]
Too bad. I only enabled encryption for the E2E (so only specific folders are encrypted, which should be possible to workaround right?), "encrypt the home storage" is not enabled. So now I have the choice to either have all my previews outside those encrypted folders generated quickly or I use E2E... too bad none of both is currently working, so not sure what to do.

To give a feedback: maybe you could at least check for a few constellations like
- "encryption enabled, default encryption module enabled, home storage not encrypted --> allow previewgenerator"
- "encryption enabled, default encryption module enabled, home storage encrypted --> don´t allow previewgenerator"
- ...

**Comment by Gerdal** [comment]
Wiem, że odgrzewam starego kotleta. Robię to ponieważ przy poszukiwaniu rozwiązań jak wygenerować miniaturki na szyfrowanych instalacjach w google - ten wątek wyskakuje jako jeden z pierwszych. Z tego co widzę, przy instalacjach nextcloud z master key encryption (czyli praktycznie wszystkie nowe instalacje) aplikacja działa poprawnie i bez problemu generuje miniaturki z zaszyfrowanych plików graficznych. Jedyne co trzeba zrobić to... zakomentować zabezpieczenie przed uruchomienieniem skryptu gdy szyfrowanie jest włączone. 

apps/previewgenerator/lib/Command/Generate.php:

```
        protected function execute(InputInterface $input, OutputInterface $output): int {
#               if ($this->encryptionManager->isEnabled()) {
#                       $output->writeln('Encryption is enabled. Aborted.');
#                       return 1;
#               }
```

apps/previewgenerator/lib/Command/PreGenerate.php:

```
        protected function execute(InputInterface $input, OutputInterface $output) {
#               if ($this->encryptionManager->isEnabled()) {
#                       $output->writeln('Encryption is enabled. Aborted.');
#                       return 1;
#               }
```

Oczywiście nie biorę za to odpowiedzialności i każdy robi to na własne ryzyko! W każdym razie - u mnie działa. Być może przydałoby się tu sprawdzenie czy master key jest włączony - i jeśli tak, dopuszczać możliwość uruchomienia skryptu.

**Comment by matttbe** [comment]
@Gerdal : thanks for this comment! It seems working for me!

Just be careful that all these previews will be written without encryption in your nextcloud data dir, in `appdata_xxxx/preview/`.

**Comment by bcutter** [comment]
As this seems to definitely NOT been fixed I´m looking forward to the fix by https://github.com/rullzer/previewgenerator/issues/156.

**Comment by Gymnae** [comment]
I've created a tiny little shell script which runs as part of spinning up a container extending the official Nextcloud FPM container: https://github.com/Gymnae/docker-owncloud/blob/fpm-noalpine/tweaks.sh
In it, I run sed to comment out the checks:
```
# add hacks to enable preview generation and proper cron jobs even if encryption is enabled
sed -i '\|this->encryptionManager->isEnabled|,/}/ s/^#*/#/' /var/www/html/custom_apps/previewgenerator/lib/Command/PreGenerate.php
sed -i '\|this->encryptionManager->isEnabled|,/}/ s/^#*/#/' /var/www/html/custom_apps/previewgenerator/lib/Command/Generate.php
sed -i '\|this->encryptionManager->isEnabled|,/}/ s/^#*/#/' /var/www/html/apps/previewgenerator/lib/Command/PreGenerate.php
sed -i '\|this->encryptionManager->isEnabled|,/}/ s/^#*/#/' /var/www/html/apps/previewgenerator/lib/Command/Generate.php
```
Preview generation will work again, but I did notice some oddities:

1. Trying to open an image uploaded via web interface to an encrypted external storage fails, preview generator also fails to generate a preview
2. Trying to upload an image with the iOS app to an external storage or opening an uploed image via web interface works flawlessly
3. Previews for images uploaded to encrypted storage are also generated by the memories app

In summary: This app here behaves weird when meeting images on encrypted external storage
5. When accessing the external
